app/assets/stylesheets/elements/_buttons.scss

```diff
@@ -137,4 +137,25 @@
   .spinner {
     margin-bottom: $margin / 8;
   }
+
+  &--icon-square {
+    @extend .button--square;
```

Comment:
Extend should only be used when applying the class itself doesn't make sense or is not easily possible. For example if you have an element `.something__child`, if you want to reuse that in a block `.else__child`, you wouldn't be able to add `something__child`, because that would be a different block. You would also only use `@extend` if it the styling has to be the exact same in all cases. `@extend` create ugly selector stacks that would be good to avoid where possible.

In this case you should apply the class `button--square` separately. But in that case the name `icon-square` doesn't make sense, since it's the  class `buton--square` that does the squaring. So just `icon` might suffice.